api/v1alpha1/fenceagentsremediationtemplate_validator.go

```diff
@@ -0,0 +1,316 @@
+
+	skipStatusValidation := false
+	// If no node-specific parameters, validate with shared parameters only, use a dummy placeholder for node name
+	if len(nodeNames) == 0 {
+		webhookTemplateValidatorLog.Info("validateFenceAgentTemplate no nodes found")
+		nodeNames["temp-validation"] = true
+		// Status validation will NOT occur for shared params with a node template (because we want to avoid getting all the nodes from the API server)
+		skipStatusValidation = true
+	}
+	// Validate parameters for each node mentioned in NodeParameters
+	for nodeName := range nodeNames {
+		// Create a temporary FAR CR from the template for this specific node
+		tempFAR := &FenceAgentsRemediation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      nodeName,
+				Namespace: r.Namespace,
+			},
+			Spec: *spec,
+		}
+
+		// BuildFenceAgentParams handles secret collection and validation internally
+		completeParams, _, err := BuildFenceAgentParams(ctx, v.Client, tempFAR)
+		if err != nil {
+			// If BuildFenceAgentParams fails, return the validation error
+			return warnings, err
+		}
+
+		if !skipStatusValidation {
+			// Validate the complete parameter set with status command
+			result := validateParametersWithStatus(spec.Agent, completeParams, v.commandExecutor)
+			if !result.IsSuccessful {
+				return warnings, fmt.Errorf("fence agent parameter validation failed: %s", result.Message)
+			}
+			// Check if successful but has a warning message
+			if result.IsSuccessful && result.Message != "" {
+				warning := fmt.Sprintf("fence agent parameter validation succeeded with warning for node %s: %s", nodeName, result.Message)
+				warnings = append(warnings, warning)
+				webhookTemplateValidatorLog.Info("validateFenceAgentTemplate warning", "node", nodeName, "warning", warning)
+			}
+		}
+	}
+	return warnings, nil
+}
+
+func getNodeNamesFromSpec(spec *FenceAgentsRemediationSpec) map[string]bool {
+	nodeNames := make(map[string]bool)
+	for _, nodeMap := range spec.NodeParameters {
+		for nodeName := range nodeMap {
+			nodeNames[string(nodeName)] = true
+		}
+	}
+	for nodeName, _ := range spec.NodeSecretNames {
+		nodeNames[string(nodeName)] = true
+	}
+	return nodeNames
+}
+
+// validateFenceAgentParams validates all fence agent parameters without building the map
+func validateFenceAgentParams(
+	far *FenceAgentsRemediation,
+	secretParams map[string]string,
+	nodeName string,
+) error {
+	// Track parameter names for uniqueness validation
+	existingParams := make(map[ParameterName]bool)
+
+	// Extract parameters from FAR
+	sharedParameters := far.Spec.SharedParameters
+	nodeParameters := far.Spec.NodeParameters
+
+	// Validate shared parameters
+	for paramName, paramVal := range sharedParameters {
+		// Verify action must be reboot
+		if err := validateActionParameter(string(paramName), paramVal); err != nil {
+			return err
+		}
+		// Verify param isn't already defined
+		if existingParams[paramName] {
+			err := errors.New(errorParamDefinedMultipleTimes)
```

Comment:
We might want to add the duplicated parameter name also in the error, as it will appear in the user's terminal and not only in the logs.